src/fhirtypes/StructureDefinition.ts

```diff
@@ -148,35 +148,37 @@ export class StructureDefinition {
     let fhirPathString = this.type;
     let matchingElements = this.elements;
     let newMatchingElements: ElementDefinition[] = [];
+    let unfoldedElements: ElementDefinition[] = [];
```

Comment:
Does `unfoldedElements` really need to be in this scope (outside of the `for` loop)?  It seems like this might cause some unintentional behavior because (unlike `newMatchingElements`) it is not guaranteed to be re-assigned on each iteration of the `for` loop -- so it seems it is theoretically possible it will be assigned during one iteration of the loop and used in another.  Even if that's not possible or likely, putting it in the smaller scope still seems like the right thing to do.